api/src/router/auth.router.ts

```diff
@@ -0,0 +1,136 @@
+import express, { Request, Response, Router } from 'express';
+import database from '../database';
+import { User } from '@prisma/client';
+import bcrypt from 'bcrypt';
+import { auth, CustomRequest } from '../middleware/auth.middleware';
+import jwt from 'jsonwebtoken';
+
+export const authRouter: Router = (() => {
+	const router = express.Router();
+
+	router.post('/register', async (req: Request, res: Response) => {
+		const user: Partial<User> = req.body;
```

Comment:
Der user sollte eigentlich keine weitere Typisierung brauchen weil diese aus dem Request kommt. Würde auch vielleicht von unnötiger Typisierung absehen damit wir nicht in Fehler laufen.